packages/terra-time-input/src/TimeUtil.js

```diff
@@ -261,9 +261,9 @@ class TimeUtil {
    */
   static getVariantFromLocale(props) {
     const { intl, variant } = props;
-    const locale12Hr = ['en-AU', 'en-CA', 'en-GB', 'en-US', 'en', 'es-US'];
+    const locale24Hr = ['de', 'es-ES', 'es', 'fi-FI', 'fr-FR', 'fr', 'nl-BE', 'nl', 'pt-BR', 'pt', 'sv-SE', 'sv'];
```

Comment:
This will lock the date-time picker to only support terra-defined locales. If someone wants to implement a custom locale, this might  not be correct. is that okay?

Reply:
So should this be reverted back to 12 hr list, or remove list completely and use empty translation logic as used previously. Your thoughts @benbcai ?

Reply:
This logic is saying that the locales defined in this locale are the ones we know that should only use 24-hour notation. All other locales (custom or the other ones we support i.e. en, en-US, en-AU, etc) will continue to honor the variant however it is set.